content/en/blog/_posts/protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,292 @@
+
+3. Assume you have a microservices architecture that includes Kafka, Zookeeper, MongoDB, a couple of jobs, a couple of CronJobs, and around 40+ microservices. Assume that if we are caught up in this situation of resource crunch and somehow Kafka or MongoDB gets evicted, then all our microservices will be useless until and unless these pods are up and start serving traffic again. 
+
+There can also be other scenarios where you want to control the order of scheduling or order of eviction of pods.
+
+## PriorityClasses in Kubernetes
+
+PriorityClass is a cluster-wide API in Kubernetes, and is a part of the `scheduling.k8s.io/v1` API group. It contains a mapping of the PriorityClass name (defined in `.metadata.name`) and an integer value (defined in `.value`). This represents the value that the scheduler uses to determine Pods' relative priority.
+
+Do you think Kubernetes itself uses PriorityClasses? The answer is yes. When you create a cluster using kubeadm, or even if you use a managed Kubernetes service (for example, Azure Kubernetes Service), PriorityClasses are widely used to safeguard the pods which are hosted on the control plane nodes. This aims to ensure that critical cluster components such as CoreDNS and kube-proxy are able to run even if resources are constrained.
+
+Using a special PriorityClass helps to make sure that these pods are up and alive to serve the requests and make sure that the overall cluster is up and running.
+
+```console
+$ kubectl get priorityclass
+NAME                      VALUE        GLOBAL-DEFAULT   AGE
+system-cluster-critical   2000000000   false            82m
+system-node-critical      2000001000   false            82m
+```
+
+The below diagram shows exactly how it works and also contains the details of the example, which will come in the upcoming section (PriorityClass HandsOn Example).
+
+{{< figure src=decision-tree.svg alt="Whenever there is a new incoming pod, this is how kube-scheduler decides their priority and puts them in a queue" title="Decision Tree">}}
+
+### Pod priority and preemption
+
+Pod preemption is a Kubernetes feature that allows the cluster to [preempt](/docs/concepts/scheduling-eviction/pod-priority-preemption/#preemption) lower-priority pods over high-priority pods in the event that there aren't enough resources to run all current Pods. To learn more about this concept, please refer to [Pod Priority and Preemption](/docs/concepts/scheduling-eviction/pod-priority-preemption/) in the Kubernetes documentation.
+
+Pod priority and preemption are Kubernetes features:
```

Comment:
We're repeating ourselves here again without adding anything new for the first, second, and last bullet points. I suggest that we either retain the previous paragraph or expand this one to cover all the features of priority & premption.

Reply:
Sure, that make sense.

```
[Pod preemption](/docs/concepts/scheduling-eviction/pod-priority-preemption/#preemption) is a Kubernetes feature that allows the cluster to preempt (evict) pods on the basis of priority and [Pod priority](/docs/concepts/scheduling-eviction/pod-priority-preemption/#pod-priority) indicates the importance of a pod relative to other pods while scheduling. If there aren't enough resources to run all the current pods, the scheduler tries to evict lower-priority pods over high-priority ones.
```

Update the above paragraph and bullet points with above paragraph.